### DISCUSSION TITLE
[transparent mode] Previously unseen connection from proxy error

### DISCUSSION THREAD

**Discussion by iscas-zac** [mitmproxy/mitmproxy#6384] [Q&A] [answered]
I'm running mitmproxy transparent mode on Windows admin Powershell. It indeed captured all traffic as I hope, but every connection timed out. I went to watch the code

https://github.com/mitmproxy/mitmproxy/blob/c851a055c30a50ecb2aeca01ae6e727464474d3d/mitmproxy/platform/windows.py#L522

and think there was a `KeyError` in my case. I have searched here and nothing is mostly relevent, so I guess it is OK to open a topic here.

By the way, my use case is originally capturing TCP/TLS traffic of a game, but I'm just learning how to use mitmproxy now. It seems that only transparent mode can capture all the traffic without additional setting for the game and DNS hosts?

**Comment by mhils** [comment, marked as answer]
This has no documentation at all yet, but with mitmproxy 10.1 you can do `mitmproxy --mode osproxy` or `mitmproxy --mode osproxy:program.exe`. You then want to take a look at the event log and see what's going on, it's not unlikely that the client does not trust the mitmproxy CA.

**Reply by iscas-zac** [reply]
That command (`mitmproxy --mode osproxy`) prints a lot of traffic, and some log entries are yellow and saying they don't trust the CA indeed (like tailscale.com and MS service)
```
 Client TLS handshake failed. The client does not trust the proxy's certificate for assets.msn.cn (OpenSSL Error([('SSL routines', '', 'sslv3 alert certificate unknown')]))
```
but most connections are established correctly, and I can browse webpages then. That is very different from the transparent case where no traffic is present at all.

Is the osproxy also a global proxy, with which I can sit in the middle of my game client and its server with no extra tweak?

**Reply by mhils** [reply]
osproxy intercepts all traffic on the local machine, or only a specific executable if you tell it to (see second invocation above). It will replace the transparent mode functionality on Windows.

**Reply by iscas-zac** [reply]
That sounds nice. I'll give it a try!

**Reply by iscas-zac** [reply]
It works perfectly! I thought I would have to write some scripts to modify the default behavior at least, but this `osproxy` mode exactly did the work. I even don't need Wireshark any more.

I notice, however, that when I click the Options -> Edit Options in the UI of `mitmweb --mode osproxy`, it prints "mitmproxy has crashed", despite that it recovered after a fresh (so it did not crash actually). I'm not sure if I should post it as an issue.

**Reply by iscas-zac** [reply]
By the way, I'm a little confused with your document page. For example, I would expect to see an example command to start up mitmproxy in [protocol](https://docs.mitmproxy.org/stable/concepts-protocols/) and [tls](https://docs.mitmproxy.org/stable/howto-wireshark-tls/) page. It would also be more friendly to have a search bar there.

Of course it takes a lot of effort, and I really appreciate your work already. It is a piece of personal opinion after all.

**Reply by mhils** [reply]
> I notice, however, that when I click the Options -> Edit Options in the UI of mitmweb --mode osproxy, it prints "mitmproxy has crashed", despite that it recovered after a fresh (so it did not crash actually). I'm not sure if I should post it as an issue.

Thanks - that's already being addressed in https://github.com/mitmproxy/mitmproxy/pull/6386. :)

> By the way, I'm a little confused with your document page. For example, I would expect to see an example command to start up mitmproxy in [protocol](https://docs.mitmproxy.org/stable/concepts-protocols/) and [tls](https://docs.mitmproxy.org/stable/howto-wireshark-tls/) page. It would also be more friendly to have a search bar there.

I agree that our docs still need some work here - that is likely to improve a bit once osproxy mode becomes an official feature. :)
